### PR TITLE
Add filename parameter to client image proxy.

### PR DESCRIPTION
**Pull request by daspecster**
Towards #2753 

Essentially I forgot to add this to the `client.image()` proxy.

### CODE REVIEW

File: vision/unit_tests/test_client.py [hunk 1]
```diff
@@ -73,13 +73,27 @@ def test_face_annotation(self):
...
+        raw_image = client.image(content=IMAGE_CONTENT)
+        self.assertIsInstance(raw_image, Image)
+        self.assertEqual(raw_image.content, B64_IMAGE_CONTENT)
+
+        with patch('google.cloud.vision.image.open',
+                   mock_open(read_data=IMAGE_CONTENT)) as m:
+            file_image = client.image(filename='my_image.jpg')
+        m.assert_called_once_with('my_image.jpg', 'rb')
+        self.assertIsInstance(file_image, Image)
+        self.assertEqual(file_image.content, B64_IMAGE_CONTENT)
```
Comment (tseaver) [inline vision/unit_tests/test_client.py]: I would split this testcase into three separate methods.

Comment (daspecster) [inline vision/unit_tests/test_client.py]: SGTM
https://github.com/GoogleCloudPlatform/google-cloud-python/pull/2792/commits/20f0f7f0dceb5603a3eca48df9a8cb388846eb6c

**Comment by daspecster** [comment]
I got a 👍  from @tseaver  on the change so I'm going to merge this.

**Comment by daspecster** [comment]
Squashed and cancelled travis. Once CircleCI goes green I'll merge.

**Pull request merged**